source/SH3/graphics/texture.cpp

```diff
@@ -13,48 +13,330 @@
+    std::ofstream file("output.tga", std::ios::binary); // Open the stream for binary output
+
+    memset(&header, 0, sizeof(header));
+
+    header.id_size      = 0x00;
+    header.colormap     = 0x00;
+    header.type         = 2;
+
+    header.x_origin     = 0x0000;
+    header.y_origin     = 0x0000;
+    header.width        = (uint16_t)width;
+    header.height       = (uint16_t)height;
+    header.bpp          = 24;
+    header.flags        = 0x20; // Origin is upper left hand corner
+
+    file.write((char*)&header, sizeof(header));
+    file.write((char*)&data[0], data.size());
+}
+
 //TODO: Scale the texture and then
 void sh3_texture::Load(sh3_arc& mft, const std::string& filename)
 {
     sh3_texture_header          header;
     sh3_arc_vfile               file(mft, filename);
     sh3_arc_vfile::read_error   e;
-    std::vector<std::uint8_t>   data;     // Pixel data of this texture (with the header stripped)
+    std::vector<uint8_t>        data;       // Pixel data of this texture (with the header stripped)
+    std::vector<rgba>           palette;    // Palette Data (I think this is BGRA)
+    std::size_t                 ret;        // How many bytes we get in from file.ReadData( )
 
-    std::size_t ret = file.ReadData(&header, sizeof(header), e); // Read in the header
+    ret = file.ReadData(&header, sizeof(header), e);
 
     if(ret != sizeof(header))
-        die("sh3_texture::Load( ): Header only read %d bytes of data! It should have been %d", ret, sizeof(header)); // TODO: Fallback on dummy texture (like Source's purple and black checkboard)
+        die("sh3_texture::Load( ): ret != sizeof(header)!");
 
-    if(header.bpp == 8)
-        Log(LogLevel::WARN, "WARNING: 8-bit textures are currently unsupported! They WILL look like nothing but pixel garbage.");
+    // Check for the pesky 64-byte A7A7A7A7 header that sometimes precedes our texture header
+    if(header.batchHeaderMarker == 0x00000000 && header.batchSize == 0xA7A7A7A7) // AHA!
+    {
+        offset = HEADER_START;
 
-    Log(LogLevel::INFO, "Texture info:\n\ttexWidth:%d\n\ttexHeight:%d\n\ttexSize:%d\n\tbpp:%d", header.texWidth, header.texHeight, header.texSize, header.bpp);
+        file.Seek(offset, std::ios_base::beg);
+        file.ReadData(&header, sizeof(header), e);
 
-    file.Seek(sizeof(header) + header.dataOffset, std::ios_base::beg);
+        if(ret != sizeof(header))
+            die("sh3_texture::Load( ): ret != sizeof(header)!");
+    }
 
-    data.resize(header.texSize);
+    // Now that we're done that, we can check perform some sanity checks on our texture!
+    if(header.texSize != header.texWidth * header.texHeight * (header.bpp / 8))
+    {
+        Log(LogLevel::WARN, "sh3_texture::Load( ): Warning, texSize != width * height * (bpp / 8)!");
+        return; // TODO: Bind a color shader here
+    }
 
-    if(header.texSize != header.texWidth * header.texHeight * header.bpp)
+    if(header.bpp == PixelFormat::PALETTE)
     {
```

Comment:
Probably. I just thought it was easier as we already had all the data we needed without having to call anything else.